src/shared/messaging/port-rpc.ts

```diff
@@ -285,14 +290,11 @@ export class PortRPC<OnMethod extends string, CallMethod extends string>
    *
    * All handlers must be registered before {@link connect} is invoked.
    */
-  on<Handler extends (...args: any[]) => void>(
-    method: OnMethod | 'close' | 'connect',
-    handler: Handler,
-  ) {
+  on<M extends keyof Handlers>(method: M, handler: Handlers[M]) {
     if (this._port) {
       throw new Error('Cannot add a method handler after a port is connected');
     }
-    this._methods.set(method, handler as any);
+    this._methods.set(method as string, handler as any);
```

Comment:
I reckon you could define `M extends keyof Handlers & string` and then you would not need to add the `as string` casting here, but I'm not 100% sure.

Reply:
It turns out the `as string` and `as any` casts are redundant with the final version of the code, so I just removed them.